python/ray/experimental/sgd/pytorch/test/pytorch_distributed_sgd_test.py

```diff
@@ -0,0 +1,33 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import numpy as np
+import torch
+
+import ray
+
+from ray.experimental.sgd.pytorch.utils import Resources
+from ray.experimental.sgd.pytorch.pytorch_distributed_sgd import PyTorchDistributedSGD
+from ray.experimental.sgd.pytorch.test.test_utils import model_creator, data_creator, optimizer_creator
+
+ray.init()
+
+sgd1 = PyTorchDistributedSGD(model_creator,
+                             data_creator,
+                             optimizer_creator,
+                             num_replicas=3)
+print(sgd1.train())
```

Comment:
Running this line on my laptop, I saw

```
2019-05-16 14:09:42,055 ERROR worker.py:1616 -- Possible unhandled error from worker: ray_PyTorchRunner:setup() (pid=6508, host=dhcp-45-39.EECS.Berkeley.EDU)
  File "/Users/rkn/Workspace/ray/python/ray/experimental/sgd/pytorch/pytorch_runner.py", line 65, in setup
    self.setup_distributed_pytorch(url, world_rank, world_size)
  File "/Users/rkn/Workspace/ray/python/ray/experimental/sgd/pytorch/pytorch_runner.py", line 76, in setup_distributed_pytorch
    dist.init_process_group(backend=self.backend,
AttributeError: module 'torch.distributed' has no attribute 'init_process_group'
```

from https://github.com/pytorch/examples/issues/467, I see that this is not supported on MacOS. What's the best way to handle the MacOS case?

Reply:
Can we raise an exception linking to the issue for now?

Reply:
Sure, I'd suggest doing it in the constructor of `PyTorchDistributedSGD`.